engine/table/src/main/java/io/deephaven/engine/table/impl/select/WhereFilterSerialImpl.java

```diff
@@ -0,0 +1,177 @@
+//
+// Copyright (c) 2016-2025 Deephaven Data Labs and Patent Pending
+//
+package io.deephaven.engine.table.impl.select;
+
+import io.deephaven.engine.liveness.LivenessArtifact;
+import io.deephaven.engine.rowset.WritableRowSet;
+import io.deephaven.engine.table.Table;
+import io.deephaven.engine.table.TableDefinition;
+import io.deephaven.engine.rowset.RowSet;
+import io.deephaven.engine.table.impl.BaseTable;
+import io.deephaven.engine.table.impl.DependencyStreamProvider;
+import io.deephaven.engine.table.impl.QueryCompilerRequestProcessor;
+import io.deephaven.engine.updategraph.NotificationQueue;
+import io.deephaven.util.SafeCloseable;
+import org.jetbrains.annotations.NotNull;
+import org.jetbrains.annotations.VisibleForTesting;
+
+import java.util.List;
+import java.util.Objects;
+import java.util.stream.Stream;
+
+/**
+ * A serial wrapper for a {@link WhereFilter} that enforces serial (non-parallel) execution.
+ */
+public class WhereFilterSerialImpl
```

Comment:
An abstract base class shared between this and io.deephaven.engine.table.impl.select.WhereFilterInvertedImpl would make sense (really for anything that is a WhereFilter wrapping another WhereFilter which we might need more of later).